pkg/utils/common.go

```diff
@@ -209,3 +217,79 @@ func NonRepeatStringsEqual(a, b []string) bool {
 	}
 	return true
 }
+
+type session struct {
+	sessionctx.Context
+	vars   *variable.SessionVars
+	values map[fmt.Stringer]interface{}
+
+	mu sync.RWMutex
+}
+
+// GetSessionVars implements the sessionctx.Context interface.
+func (se *session) GetSessionVars() *variable.SessionVars {
+	return se.vars
+}
+
+// SetValue implements the sessionctx.Context interface.
+func (se *session) SetValue(key fmt.Stringer, value interface{}) {
+	se.mu.Lock()
+	se.values[key] = value
+	se.mu.Unlock()
+}
+
+// Value implements the sessionctx.Context interface.
+func (se *session) Value(key fmt.Stringer) interface{} {
+	se.mu.RLock()
+	value := se.values[key]
+	se.mu.RUnlock()
+	return value
+}
+
+// UTCSession can be used as a sessionctx.Context, with UTC timezone.
+var UTCSession *session
+
+func init() {
+	UTCSession = &session{}
+	vars := variable.NewSessionVars()
+	vars.StmtCtx.TimeZone = time.UTC
+	UTCSession.vars = vars
+	UTCSession.values = make(map[fmt.Stringer]interface{}, 1)
+}
+
+// AdjustBinaryProtocolForDatum converts the data in binlog to TiDB datum.
+func AdjustBinaryProtocolForDatum(data []interface{}, cols []*model.ColumnInfo) ([]types.Datum, error) {
+	log.L().Debug("AdjustBinaryProtocolForChunk",
+		zap.Any("data", data),
+		zap.Any("columns", cols))
+	ret := make([]types.Datum, 0, len(data))
+	for i, d := range data {
+		switch v := d.(type) {
+		case int8:
+			d = int64(v)
+		case int16:
+			d = int64(v)
+		case int32:
+			d = int64(v)
+		case uint8:
+			d = uint64(v)
+		case uint16:
+			d = uint64(v)
+		case uint32:
+			d = uint64(v)
+		case uint:
+			d = uint64(v)
+		case decimal.Decimal:
+			d = v.String()
```

Comment:
Better to add a default branch.

Reply:
what to do in default branch? `d` is directly used other than above types.